code/modules/mob/living/basic/lavaland/goliath/goliath.dm

```diff
@@ -0,0 +1,225 @@
+	AddElement(/datum/element/footstep, FOOTSTEP_MOB_HEAVY)
+	AddElement(/datum/element/basic_eating, heal_amt = 10, food_types = goliath_foods)
+	AddElement(/datum/element/move_cooldown, move_delay = movement_delay)
+	AddComponent(/datum/component/basic_mob_attack_telegraph)
+	AddComponentFrom(INNATE_TRAIT, /datum/component/shovel_hands)
+	if (tameable)
+		AddComponent(/datum/component/tameable, food_types = list(/obj/item/food/grown/ash_flora), tame_chance = 10, bonus_tame_chance = 5, after_tame = CALLBACK(src, PROC_REF(tamed)))
+
+	tentacles = new (src)
+	tentacles.Grant(src)
+	var/datum/action/cooldown/tentacle_burst/melee_tentacles = new (src)
+	melee_tentacles.Grant(src)
+	AddComponent(/datum/component/revenge_ability, melee_tentacles, targetting = ai_controller.blackboard[BB_TARGETTING_DATUM], max_range = 1, target_self = TRUE)
+	var/datum/action/cooldown/tentacle_grasp/ranged_tentacles = new (src)
+	ranged_tentacles.Grant(src)
+	AddComponent(/datum/component/revenge_ability, ranged_tentacles, targetting = ai_controller.blackboard[BB_TARGETTING_DATUM], min_range = 2, max_range = 9)
+
+	tentacles_ready()
+	RegisterSignal(src, COMSIG_MOB_ABILITY_FINISHED, PROC_REF(used_ability))
+	ai_controller.set_blackboard_key(BB_BASIC_FOODS, goliath_foods)
+	ai_controller.set_blackboard_key(BB_GOLIATH_TENTACLES, tentacles)
+
+/mob/living/basic/mining/goliath/Destroy()
+	QDEL_NULL(tentacles)
+	return ..()
+
+/mob/living/basic/mining/goliath/examine(mob/user)
+	. = ..()
+	if (saddled)
+		. += span_info("Someone appears to have attached a saddle to this one.")
+
+/mob/living/basic/mining/goliath/revive(full_heal_flags, excess_healing, force_grab_ghost)
+	. = ..()
+	if (!.)
+		return
+	move_force = initial(move_force)
+	move_resist = initial(move_resist)
+	pull_force = initial(pull_force)
+
+/mob/living/basic/mining/goliath/death(gibbed)
+	move_force = MOVE_FORCE_DEFAULT
+	move_resist = MOVE_RESIST_DEFAULT
+	pull_force = PULL_FORCE_DEFAULT
+	return ..()
+
+// Goliaths can summon tentacles more frequently as they take damage, scary.
+/mob/living/basic/mining/goliath/apply_damage(damage, damagetype, def_zone, blocked, forced, spread_damage, wound_bonus, bare_wound_bonus, sharpness, attack_direction, attacking_item)
+	. = ..()
+	if (!.)
+		return
+	if (damage <= 0)
+		return
+	if (tentacles.cooldown_time > 1 SECONDS)
+		tentacles.cooldown_time -= 1 SECONDS
+
+/mob/living/basic/mining/goliath/attackby(obj/item/attacking_item, mob/living/user, params)
+	if (!istype(attacking_item, /obj/item/goliath_saddle))
+		return ..()
+	if (!tameable)
+		balloon_alert(user, "doesn't fit!")
+		return
+	if (saddled)
+		balloon_alert(user, "already saddled!")
+		return
+	if (!tamed)
+		balloon_alert(user, "too rowdy!")
+		return
+	balloon_alert(user, "affixing saddle...")
+	if (!do_after(user, delay = 5.5 SECONDS, target = src))
+		return
+	user.visible_message(span_notice("[user] mounts [attacking_item] to [src], [p_they()] are ready for riding!"))
```

Comment:
this could also be something like `balloon_alert("ready to ride")`? up to you though

Reply:
Loses visibility to other people nearby but the sprite visibly changes so that's fine.